RATIONALE.adoc

```diff
@@ -71,32 +71,15 @@ Hopefully you’ll be able to see these principles reflected in the new API.
 
 In this section we’ll outline the major changes, and most prominent points of interest in the new API.
 
-=== An explicit concept of ownership
-
-In the new API we make it explicit that a buffer can have an “owner”.
-This is the case when the buffer isn’t borrowed out anywhere else; the reference count is one.
-In this state the buffer is said to be “owned”.
-This is important because certain operations can only be called when the buffer is in an owned state, as they could otherwise violate our safety requirements.
-All of these operations will call this out in their javadocs.
-These operations either change the ownership of the buffer, or they change the internal memory allocation of the buffer, in some way.
-For instance, `ensureWritable()` requires ownership because it may replace the internal memory allocation with a new larger one.
-The `split()` method requires ownership because it changes how the buffer ownership itself is set up.
-The `send()` method is used for transferring ownership between threads, which obviously requires ownership to begin with.
-
-[source,java]
-----
-if (buf.isOwned()) {
-    buf.compact(); // compact() requires ownership because it moves data.
-}
-----
-
 === Reference counting
 
 Buffers are now `AutoCloseable` and can be used in try-with-resources clauses.
-Every allocation and acquire call on a buffer (any `Rc` object, really) must be paired with a `close()`, and every `receive()` call on a `Send` must also be paired with a `close()`.
-Every method that increments the reference count will document this fact.
-When the reference count is not incremented, then there is obviously no need to decrement it either.
-Buffers can thus be passed through a pipeline without having to do much reference counting work.
+Every allocation and acquire call on a buffer (any `Resource` object, really) must be paired with a `close()`, and every `receive()` call on a `Send` must also be paired with a `close()`.
+
+While referene counting is a useful thing for tracking resource life-cycles internally, it is not itself exposed in the public API.
+Instead, the public API effectively has a boolean open/closed state.
+This simplifies the API a great deal; buffers are created, and in the end they are closed.
+The code in between needs to be arranged such that it just avoids ever holding on to buffers that might be closed.
```

Comment:
I like this... this also removes the pattern where sometimes people do something like:

```
while (buf.refCnt() > 0) {
    buf.release();
}
```
I never saw a use-case where this not just did hide a bug somewhere else.